tests/extensions/zip/test_renderer.py

```diff
@@ -0,0 +1,61 @@
+import os
+import pytest
+from zipfile import ZipFile
+
+from mfr.core.provider import ProviderMetadata
+
+from mfr.extensions.zip import ZipRenderer
+
+BASE = os.path.dirname(os.path.abspath(__file__))
+
+
+@pytest.fixture
+def metadata():
+    return ProviderMetadata('test', '.zip', 'text/plain', '1234', 'http://wb.osf.io/file/test.zip?token=1234')
+
+@pytest.fixture
+def zip_file():
+    return ZipFile(os.path.join(BASE, 'files', 'test.zip'), 'r')
+
+
+@pytest.fixture
+def test_file_path():
+    return os.path.join(os.path.dirname(os.path.abspath(__file__)), 'files', 'test.zip')
+
+@pytest.fixture
+def test_render_body():
+    with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), 'files', 'test_render_body.html'), 'r') as fp:
+        return fp.read()
+
+
+@pytest.fixture
+def url():
+    return 'http://osf.io/file/test.zip'
+
+
+@pytest.fixture
+def assets_url():
+    return 'http://mfr.osf.io/assets'
+
+
+@pytest.fixture
+def export_url(url):
+    return 'http://mfr.osf.io/export?url=' + url
+
+
+@pytest.fixture
+def renderer(metadata, test_file_path, url, assets_url, export_url):
+    return ZipRenderer(metadata, test_file_path, url, assets_url, export_url)
+
+
+class TestZipRenderer:
+
+    def test_render(self, renderer):
+        body = renderer.render()
+
+        assert '<td>test 2' in body and '<td>test 1' in body
+
+    def test_format_zip(self, renderer, zip_file):
+        zip_string = renderer.format_zip(zip_file)
+
+        assert '<td>test 2' in zip_string and '<td>test 1' in zip_string
```

Comment:
Please expand your testing to match the expanded functionality of the zip renderer. Things I'd like to see:

* the timestamps are present and correct
* the byte size is present and correct
* proper formatting of larger byte sizes 1k, 1M, etc.  (for 1M, a megabyte text file of a single character should compress pretty well)
* `__MACOSX` files are actually being excluded
* What happens when you pass in something that isn't a zipfile?  Is the error message informative or vague?
* Is it possible to have an empty zipfile?  If so, we should state that instead of displaying an empty table.
* zipped zipfiles should *not* be recursively expanded.